File: packages/sdk/js-example/src/batch-async-wait-item.ts
import type { CompletionBatchRequest } from '@vm-x-ai/sdk';
import { VMXClient } from '@vm-x-ai/sdk';
import axios from 'axios';

const main = async () => {
  const client = new VMXClient();
  console.log('Starting batch');

  const requests = Array(10)
    .fill(0)
    .map<CompletionBatchRequest['requests'][number]>(() => ({
      messages: [
        {
          role: 'user',
          content: 'Hi',
        },
      ],
      resource: 'default',
    }));

  const response = await client.completionBatch(requests);

  console.log('Batch created', response.batchId);

  for (const item of response.items) {
    const result = await client.waitForCompletionBatchItem(response.batchId, item.itemId);
    console.log('Batch item', result.response?.message);
  }
};

main()
  .catch((error) => {
    if (axios.isAxiosError(error)) {
      console.error(error.request);
      console.error(error.response?.data);
    } else {
      console.error(error);
    }
    process.exit(1);
  })
  .finally(() => process.exit(0));

File: packages/sdk/python-example/example/batch_async_wait_item.py
#!/usr/bin/env python
import asyncio

from vmxai import VMXClient
from vmxai.types import CompletionRequest, RequestMessage


async def main():
    """Example demonstrating how to submit batch completion requests asynchronously."""
    client = VMXClient()
    print("Starting batch")

    # Create 100 completion requests
    requests = [
        CompletionRequest(
            messages=[
                RequestMessage(
                    role="user",
                    content="Hi",
                )
            ],
            resource="default",
        )
        for _ in range(10)
    ]

    # Submit the batch asynchronously
    response = await client.completion_batch(requests)
    print(f"Batch created: {response.batch_id}")

    # Get the result of a batch completion item
    for item in response.items:
        result = await client.wait_for_completion_batch_item(response.batch_id, item.item_id)
        print(f"Batch item: {result.response.message}")


if __name__ == "__main__":
    try:
        asyncio.run(main())
    except Exception as e:
        print(f"Error: {e}")
        exit(1) 